web/src/app/components/job-list-item/job-list-item.component.ts

```diff
@@ -82,18 +82,14 @@ export class JobListItemComponent implements OnInit, OnDestroy {
         this.surveyId = id;
       })
     );
-
-    // Add initial node for current job
-    this.dataSource.data = this.dataSource.data.concat([
-      new DynamicFlatNode(
-        /* name= */ this.job!.name!,
-        /* level= */ 0,
-        /* expandable= */ true,
-        /* iconName= */ 'label',
-        /* iconColo= */ this.job!.color!,
-        /* jobId= */ this.job!.id
-      ),
-    ]);
+    this.subscription.add(
+      this.loiService.getLocationsOfInterest$().subscribe(lois => {
+        console.log(this.job!.name, lois.size);
+        this.dataSource.data = this.dataSource.data.concat([
```

Comment:
Is there a way to do this without relying on side effects? Can you instead use `*ngIf=` and look at the LOI count, for example?